image/walker.go

```diff
@@ -92,11 +92,16 @@ func newPathWalker(root string) walker {
 
 func (w *pathWalker) walk(f walkFunc) error {
 	return filepath.Walk(w.root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return errors.Wrap(err, "error walking path")
+		}
+		if info == nil {
```

Comment:
I'm not sure the error message here is meaningful. I think we should omit this check (perhaps you could also file a bug upstream on golang to clarify the guarantees here)